### ISSUE TITLE
FSR2 and TAA are affecting MultiMeshes in weird ways

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#93958]
### Tested versions

Godot 4.2
Godot 4.3 beta2

### System information

Windows 10 64 bits NVIDIA GeForce GTX 1060

### Issue description

I got a report from someone who got a strange issue going on when multimesh buffers get modified, causing instances to get corrupted: https://github.com/Zylann/godot_voxel/issues/670

Investigating revealed that FSR2 seems to cause the issue. Turning it off makes the issue non-reproducible.

I tried making a simple project where I generate multimeshes and randomly edit and change their visible counts. Despite not being able to reproduce a situation that *looks* the same, I could still observe a striking difference in behavior between enabling FSR2 and not enabling it. This is not expected.

### Steps to reproduce

- Have multimeshes in your scene
- Use `get_instance_transform` and `set_instance_transform` to change specific instances
- Set `visible_instance_count` to a smaller count progressively

### Minimal reproduction project (MRP)

[MultiMeshFSR2.zip](https://github.com/user-attachments/files/16103055/MultiMeshFSR2.zip) (Godot 4.3 beta2)

This project defaults to a scaled down 3D rendering, using bilinear scaling.
- Launch main scene
- Observe several multimesh instances are showing up:

![image](https://github.com/godotengine/godot/assets/1311555/ab81dfe7-5f61-4da5-9861-60ce12188825)

- Press space: it removes a few random instances from a specific multimesh, as expected by the code

![image](https://github.com/godotengine/godot/assets/1311555/2485cfaa-9a98-4cb8-86d2-46dba38a45b8)
^ Result after 5 presses.

- Close the game
- In Project Settings, change scaling mode to use FSR2
- Launch main scene again
- Press space: observe all but one instance disappear in one go almost every time, sometimes the top half, very rarely a few... it's way too inconsistent to be expected, and the expected counts printed don't match what's rendered.

![image](https://github.com/godotengine/godot/assets/1311555/66f100aa-3dc1-4983-a548-3a61247bc5b0)
^ Result after 5 presses

Maybe the repro can be simplified further. I mostly tried to reproduce a scenario close to what happens in the full project.

**Comment by Calinou** [comment]
Does this behavior change if you change the **Threaded Cull** threshold to a much higher value in the project settings (so that threaded culling is not used)?

**Comment by Zylann** [comment]
The scene has only 16 multimesh instances, which is way below the default threshold (1000). I tried anyways to raise it to the maximum I could (65536) and that had no effect.
Also, as shown in the video of the linked issue, the problem doesn't just make multimesh instances disappear, it corrupts their transforms so sometimes they show up with random scales or flicker around the camera.

**Comment by DarioSamo** [comment]
While I can't investigate this in detail at the moment, the points of interest is any functionality that affects multimesh and the doubling in size of its transforms buffer when TAA/FSR2 is enabled, which is required for motion vectors.

```cpp
void MeshStorage::_multimesh_enable_motion_vectors(MultiMesh *multimesh)
```
```cpp
void MeshStorage::_multimesh_get_motion_vectors_offsets(RID p_multimesh, uint32_t &r_current_offset, uint32_t &r_prev_offset)
```

**Comment by Rudolph-B** [comment]
I can't replicate it in the current master branch. After investigating it appears that this was fixed with #95270.

**Comment by clayjohn** [comment]
Thanks for confirming!

Closing as fixed.
